### PR TITLE
fix: Log athena query id when it is available. Closes #530

### LINKED ISSUE #530: Log the query id
Hello! 

Thank you for making this adapter for Athena, we depend on this heavily. 

Not sure if this has already been discussed/considered. In order to help with any issues related to a particular query, it would be great if Athena Query ID get logged at some point.

I see the sql is getting logged on exception, which is great.
https://github.com/dbt-athena/dbt-athena/blob/main/dbt/adapters/athena/connections.py#L216

Since Athena provides a unique query id for each query it would be helpful to log it
- on start query execution
- on query completion
- in case of exception

Issue triaging would be much smoother with this

### PR DESCRIPTION
**Pull request by WarFox**
# Description

Adds info log as soon as query ID is available
Adds query_id to warning and debug logs in AthenaCursor to help with issue triaging

resolve #530 

## Models used to test - Optional
NA

## Checklist

- [ ] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [x] You kept your Pull Request small and focused on a single feature or bug fix.
- [x] You added unit testing when necessary
- [x] You added functional testing when necessary

### CODE REVIEW

File: dbt/adapters/athena/connections.py [hunk 1]
```diff
@@ -163,6 +163,9 @@ def inner() -> AthenaCursor:
                 cache_size=cache_size,
                 cache_expiration_time=cache_expiration_time,
             )
+
+            log.info(f"Athena query ID {query_id}")
```
Comment (nicor88) [inline dbt/adapters/athena/connections.py:167]:
```suggestion
            logger.debug(f"Athena query ID {query_id}")
```

Comment (nicor88) [reply dbt/adapters/athena/connections.py:167]: @WarFox could you please use `logger` and not log, and the use the `debug` mode.
Using info severity will lead to having the queries ids printed in the stout in `dbt run`...

Comment (WarFox) [reply dbt/adapters/athena/connections.py]: Thank you, it is done now

**Comment by nicor88** [comment]
thanks @WarFox

**Review by nicor88 — APPROVED** [review]

**Pull request merged**